Bottom: 7abff1e6f02afaad2288000d9bd122414d507aa5
Top:    4b7afcf924b93f2e624bc71cd9de4dfdb02bfe06
Author: Ian Kent <ikent@redhat.com>
Date:   2012-11-12 11:33:41 +0800

autofs-5.0.7 - disable hosts map HUP signal update


---

diff --git a/man/auto.master.5.in b/man/auto.master.5.in
index 21d7544..cacba1a 100644
--- a/man/auto.master.5.in
+++ b/man/auto.master.5.in
@@ -224,10 +224,7 @@ set default log level "none", "verbose" or "debug" (program default "none").
 .SH BUILTIN MAP -hosts
 If "-hosts" is given as the map then accessing a key under the mount point
 which corresponds to a hostname will allow access to the exports of that
-host. The hosts map cannot be dynamically updated and requires a HUP signal
-to be sent to the daemon for it to check hosts for an update. Due to possible
-hierarchic dependencies within a mount tree, it might not be completely
-updated during the HUP signal processing.
+host.
 .P
 For example, with an entry in the master map of
 .nh
diff --git a/man/autofs.5 b/man/autofs.5
index 55a8a7c..2161116 100644
--- a/man/autofs.5
+++ b/man/autofs.5
@@ -13,10 +13,10 @@ These maps describe how file systems below the mount point of the map
 map format; if another map format is specified (e.g. \fBhesiod\fP),
 this documentation does not apply.
 
-Indirect maps, except for the internal hosts map, can be changed on the fly
-and the automouter will recognize those changes on the next operation it
-performs on that map. Direct maps require a HUP signal be sent to the
-daemon to refresh their contents as does the master map.
+Indirect maps can be changed on the fly and the automouter will recognize
+those changes on the next operation it performs on that map. Direct maps
+require a HUP signal be sent to the daemon to refresh their contents as does
+the master map.
 .SH "FORMAT"
 This is a description of the text file format.  Other methods of specifying
 these files may exist.  All empty lines or lines beginning with # are
diff --git a/modules/lookup_hosts.c b/modules/lookup_hosts.c
index 8a8a34d..a40dc84 100644
--- a/modules/lookup_hosts.c
+++ b/modules/lookup_hosts.c
@@ -248,10 +248,7 @@ int lookup_read_map(struct autofs_point *ap, time_t age, void *context)
 	 * direct mounts in order to mount the triggers.
 	 */
 	if (!(ap->flags & MOUNT_FLAG_GHOST) && ap->type != LKP_DIRECT) {
-		debug(ap->logopt, MODPREFIX
-		      "map not browsable, update existing host entries only");
-		update_hosts_mounts(ap, source, age, ctxt);
-		source->age = age;
+		debug(ap->logopt, "map read not needed, so not done");
 		return NSS_STATUS_SUCCESS;
 	}
 
@@ -275,7 +272,6 @@ int lookup_read_map(struct autofs_point *ap, time_t age, void *context)
 	if (status)
 		error(ap->logopt, MODPREFIX "failed to unlock hostent mutex");
 
-	update_hosts_mounts(ap, source, age, ctxt);
 	source->age = age;
 
 	return NSS_STATUS_SUCCESS;
